quinn-proto/src/connection/qlog.rs

```diff
@@ -281,6 +285,111 @@ impl QlogSink {
             stream.emit_event_with_tuple_id(EventData::PacketReceived(event), now, tuple_id);
         }
     }
+
+    /// Emits a timer event.
+    ///
+    /// This function is not public: Instead, create a [`QlogSinkWithTime`] via [`Self::with_time`] and use
+    /// its `emit_timer_` methods.
+    #[cfg(feature = "qlog")]
+    fn emit_timer(&self, timer: Timer, op: TimerOp, now: Instant) {
+        let Some(stream) = self.stream.as_ref() else {
+            return;
+        };
+
+        let timer_type: Option<TimerType> = match timer {
+            Timer::Conn(conn_timer) => match conn_timer {
+                ConnTimer::Idle => Some(QlogTimerType::IdleTimeout.into()),
+                ConnTimer::Close => Some(TimerType::custom("close")),
+                ConnTimer::KeyDiscard => Some(TimerType::custom("key_discard")),
+                ConnTimer::KeepAlive => None,
+                ConnTimer::PushNewCid => Some(TimerType::custom("push_new_cid")),
+            },
+            Timer::PerPath(_, path_timer) => match path_timer {
+                PathTimer::LossDetection => Some(QlogTimerType::LossTimeout.into()),
+                PathTimer::PathIdle => Some(TimerType::custom("path_idle")),
+                PathTimer::PathValidation => Some(QlogTimerType::PathValidation.into()),
+                PathTimer::PathChallengeLost => Some(TimerType::custom("path_challenge_lost")),
+                PathTimer::PathOpen => Some(TimerType::custom("path_open")),
+                PathTimer::PathKeepAlive => None,
+                PathTimer::Pacing => None,
```

Comment:
Also not sure why you didn't include them? Are we worried about perf?

Reply:
Yeah, config levels are interesting. I was initially thinking that the viewer would be able to toggle some events off, so it doesn't matter much if we flood the logs (other than totally taking perf and/or file sizes i guess).  /cc @Arqu